### PR TITLE
Adding Warlock Draconic tank tier 2 / Wicked

### PR DESCRIPTION
**Pull request by OmegaHelix**
item_sets_pve: 
1. Add Warlock Tank Tier 2 2/4/6p set bonuses

Base spell changes:
Death Coil  / Drain Life: 
1. Use CalcAndDealHealing to trigger OnHealDealt callback

Lifetap: 
1. Expose Ranks / Damage values for tier 2p to hit target
2. Add SpellFlagBinary to avoid partial resists
3. Change to OutcomeAlwaysHit so it no longer crits / misses

2/3 lowers dps's dps tests by ~1, expected

action_id: 
1. Add (Heal) for clarity, case needed to avoid (??) with death coil / drain life heals

### CODE REVIEW

File: sim/warlock/death_coil.go [hunk 1]
```diff
@@ -17,6 +17,16 @@ func (warlock *Warlock) getDeathCoilBaseConfig(rank int) core.SpellConfig {
 
 	baseDamage *= 1 + warlock.shadowMasteryBonus()
 
+	healingSpell := warlock.GetOrRegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: spellId}.WithTag(1),
+		SpellSchool: core.SpellSchoolPhysical,
+		ProcMask:    core.ProcMaskSpellHealing,
+		Flags:       core.SpellFlagPassiveSpell,
```
Comment (kayla-glick) [inline sim/warlock/death_coil.go:24]: You may want to include `core.SpellFlagHelpful` here too

File: sim/warlock/drain_life.go [hunk 2]
```diff
@@ -29,7 +29,16 @@ func (warlock *Warlock) getDrainLifeBaseConfig(rank int) core.SpellConfig {
 	baseDamage *= 1 + warlock.shadowMasteryBonus() + 0.02*float64(warlock.Talents.ImprovedDrainLife)
 
 	actionID := core.ActionID{SpellID: spellId}
-	healthMetrics := warlock.NewHealthMetrics(actionID)
+
+	healingSpell := warlock.GetOrRegisterSpell(core.SpellConfig{
+		ActionID:    actionID.WithTag(1),
+		SpellSchool: core.SpellSchoolPhysical,
+		ProcMask:    core.ProcMaskSpellHealing,
+		Flags:       core.SpellFlagPassiveSpell,
```
Comment (kayla-glick) [inline sim/warlock/drain_life.go:37]: Same here

File: sim/warlock/death_coil.go (hunk 1 above)
Comment (OmegaHelix) [reply sim/warlock/death_coil.go]: added flag

File: sim/warlock/drain_life.go (hunk 2 above)
Comment (OmegaHelix) [reply sim/warlock/drain_life.go]: added flag

**Review by kayla-glick — APPROVED** [review]

**Pull request merged**